railties/lib/rails/generators/actions.rb

```diff
@@ -99,11 +99,15 @@ def gem(*args)
           end
           str << indentation
           str << "gem #{parts.join(", ")}"
-          append_file_with_newline "Gemfile", str.join, verbose: false
+          if @gem_groups
+            inject_into_file "Gemfile", "\n#{str.join}", after: "group #{@gem_groups} do"
```

Comment:
This made for the simplest implementation, but it would be nice if we append the gem group instead. I think for that, we'd need to use [gsub_file](https://www.rubydoc.info/gems/thor/Thor/Actions#gsub_file-instance_method) which is more complicated.